r/R/dplyr-functions.R

```diff
@@ -442,3 +442,37 @@ nse_funcs$strptime <- function(x, format = "%Y-%m-%d %H:%M:%S", tz = NULL, unit
 
   Expression$create("strptime", x, options = list(format = format, unit = unit))
 }
+
+nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption("lubridate.week.start", 7)) {
+  if (label) {
+    arrow_not_supported("Label argument")
```

Comment:
What is the label argument? Is it something we should support? I think it's good to document these unsupported exceptions in comments, to note either why Arrow should behave differently, or what Jira issue exists to remove the exception.

Reply:
The label argument is whether the day is displayed as a character string or a number.  Maybe it is something we should support; shall I open a ticket?

Reply:
The label argument allows the return of the day of the week in the correct language.

In `lubridate`, there's a whole lot of work around defining the locale and then setting the weekday names appropriately, though essentially, it just uses the format function to extract the relevant date.
	
I think that we wouldn't want to implement this at the C++ level as it doesn't feel like it fits there as it's doing quite a lot more work than calculating a value, (i.e. returning locale-specific dates), and we could still achieve this with good performance at the R level by implementing this functionality using the ifelse kernel once its R bindings are implemented.

How about if I leave it as unsupported now, but create a ticket to implement it later, that lists the ifelse R binding as a dependency?

Reply:
We could do such things in C++ as the library we use supports locale-aware [strftime](https://strftime.org/) (See [slide 26](http://schd.ws/hosted_files/cppcon2016/0f/Welcome%20To%20The%20Time%20Zone%20-%20Howard%20Hinnant%20-%20CppCon%202016.pdf). I expect we'll need it other places later too.

Please make a Jira and indicate we need to discuss if this goes into R or C++.

Reply:
Thanks, will do!

Reply:
https://issues.apache.org/jira/browse/ARROW-13133

Reply:
Can you leave a comment explaining this and linking to the JIRA issue?

Reply:
Yup, on it!